proxy-node-gateway/src/main/java/uk/gov/ida/notification/redis/SessionCacheManager.java

```diff
@@ -0,0 +1,51 @@
+package uk.gov.ida.notification.redis;
+
+import io.dropwizard.lifecycle.Managed;
+import uk.gov.ida.notification.redis.exceptions.SessionAlreadyExistsException;
+import uk.gov.ida.notification.redis.exceptions.SessionMissingException;
+import uk.gov.ida.notification.redis.storage.InMemoryStorage;
+import uk.gov.ida.notification.redis.storage.RedisStorage;
+import uk.gov.ida.notification.redis.storage.StorageType;
+import uk.gov.ida.notification.session.GatewaySessionData;
+
+import java.net.URI;
+
+public class SessionCacheManager implements Managed {
+
+    private StorageType storage;
+
+    public SessionCacheManager(URI redisServerURI) {
+        storage = !redisServerURI.toString().isEmpty() ?
```

Comment:
Could remove the `!` and flip the ternary options to make it slightly easier to mentally parse?